src/views/modals/ExampleFormModal.tsx

```diff
@@ -0,0 +1,130 @@
+                <div className="modalForm-item">
+                    <label
+                        htmlFor="vapor-modal-form-name"
+                        className="modalForm-item-label"
+                    >
+                        {'Name'}<sup>{'*'}</sup>
+                    </label>
+                    <div className="modalForm-item-input">
+                        <InputField
+                            id={'vapor-modal-form-name'}
+                            name="name"
+                            isRequired={true}
+                        />
+                    </div>
+                </div>
+                <div className="modalForm-item">
+                    <label
+                        htmlFor="vapor-modal-form-exposure-limit"
+                        className="modalForm-item-label"
+                    >
+                        {'Age'}<sup>{'*'}</sup>
+                    </label>
+                    <div className="modalForm-item-input">
+                        <InputField
+                            id={'example-form-age'}
+                            name="age"
+                            step={'1'}
+                            type={'number'}
+                            isRequired={true}
+                        />
+                    </div>
+                </div>
+                <div className="modalForm-item">
+                    <label
+                        htmlFor="vapor-modal-form-molecular-weight"
+                        className="modalForm-item-label"
+                    >
+                        {'Email'}<sup>{'*'}</sup>
+                    </label>
+                    <div className="modalForm-item-input">
+                        <InputField
+                            id={'example-form-modal-email'}
+                            name="email"
+                            pattern="[a-z0-9._%+-]+@[a-z0-9.-]+\.[a-z]{2,3}$"
+                            isRequired={true}
+                        />
+                    </div>
+                </div>
+                <div className="modalForm-item">
+                    <label
+                        htmlFor="example-form-modal"
+                        className="modalForm-item-label"
+                    >
+                        {'Address'}
+                    </label>
+                    <div className="modalForm-item-input">
+                        <InputField
+                            id={'example-form-modal-address'}
+                            name="address"
+                        />
+                    </div>
+                </div>
+            </form>
+        );
+    }
+
+    private _onClickAccept(event: React.MouseEvent<HTMLButtonElement>): void {
+        event.preventDefault();
+
+        if (this._formElement.checkValidity()) {
+            const formData: any = form2js(this._formElement, '.', false);
+
+            console.log(formData);
+
+            this.props.closeModal();
+        } else {
```

Comment:
this trailing `else` bothers me.  _In my opinion_ refactoring it thusly gives you two strict test cases and is easier to read.

```javascript
private _onClickAccept(event: React.MouseEvent<HTMLButtonElement>): void {
    event.preventDefault();

    if (!this._formElement.checkValidity()) {
        this._formElement.classList.remove('u-validate');

        return
    }
    
    const formData: any = form2js(this._formElement, '.', false);

    this.props.closeModal();
}
```

Reply:
I am going to keep the else

Reply:
I knew you would